### ISSUE TITLE
I met a problem, when the feature's style contains image and the number of feature is more than 32.

### ISSUE THREAD

**Issue by dupeiduo** [openlayers/openlayers#7233]
The [jsfiddle url is here](http://jsfiddle.net/KaerDu/3xq0k2ad/9/). The pictures will flash when the change event happened and the flash maybe won't stop.

I guess the problem is here.
![image](https://user-images.githubusercontent.com/15188363/30156180-8d4c6d04-93f1-11e7-8bda-28efd2fc0172.png)

Why dispatch a "change" event after each image loaded?

**Comment by ahocevar** [comment]
Don't create a new image style with every call. Cache the styles by url:
```js
var imageStyleCache = {};
function getImage(url) {
  if (!imageStyleCache[url]) {
    imageStyleCache[url] = new ol.style.Style({
      image: new ol.style.Icon({
        anchor: [0.5, 65],
        anchorYUnits: 'pixels',
        src: url,
        imgSize: [61, 55]
      })
    });
  }

  return imageStyleCache[url]
}
```
Updated fiddle: http://jsfiddle.net/KaerDu/3xq0k2ad/9/

**Comment by dupeiduo** [comment]
@ahocevar  Yes, I solved the problem like that. Will you provide a method to set the max iconImageCache?

**Comment by ahocevar** [comment]
Good idea @dupeiduo. We'd accept a pull request to add something like `ol.style.iconImageCache#setMaxCacheSize()`.

**Comment by bobobobo** [comment]
See PR #7372

**Comment by fredj** [comment]
fixed with #7372
